src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,67 @@
+package duke.command;
+
+import duke.data.TaskList;
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.ui.Ui;
+
+public abstract class AddCommand extends Command {
+    public static String COMMAND_WORD;
+    protected String taskName;
+
+    public AddCommand() {
+        super();
+    }
+
+    public abstract void checkAndSetParameters(String parameterInput) throws DukeException;
+
+    protected abstract void checkParameters(String parameterInput) throws DukeException;
+
+    protected abstract void setParameters(String parameterInput);
+
+    public abstract void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException;
+
+    protected boolean containsTaskDescription(String parameterInput, String separator) {
+        // Trim description from parameterInput
+        String[] splits = parameterInput.split(separator, 2);
+        String description = splits[0];
+
+        // Check if description is empty or only contain whitespaces
+        description = description.replaceAll("\\s+", "");
+        if (description.equals("")) {
+            return false;
+        } else {
+            return true;
+        }
+    }
```

Comment:
Might want to refactor this into a separate method, since you use it multiple times.

Reply:
Thanks for the advice!!

I've changed this part to:
```
hasDescription = !description.equals("");
return hasDescription;
```
to make it cleaner and I believe this doesn't need a refractor anymore